src/vehicle.cpp

```diff
@@ -4693,7 +4794,12 @@ veh_collision vehicle::part_collision( int part, const tripoint &p,
                 }
             }
         } else if( ret.type == veh_coll_body ) {
-            int dam = obj_dmg*dmg_mod/100;
+            int dam = obj_dmg * dmg_mod / 100;
+
+            if( is_body_collision && parts[ret.part].enabled &&
+                part_info( ret.part ).has_flag( "CHAINSAW" ) ) {
+                dam += damage_from_chainsaw( ret.part, critter->get_material() );
```

Comment:
I expect to expand this function in another direction and process colliding with walls and other vehicles with the same function so I'll keep critter reference out of it (at least for now). ```part_info( part_index ).has_flag( "CHAINSAW" )``` check was moved in.